### PR TITLE
refactor: untangling lib/index.js part 1, monitor

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?

* add a npm script to track circular imports
* replace indirect references to `lib/monitor.ts` through `snyk` (which is `lib/index.js`) with direct imports
* tidy-up monitor.ts module

see more in Background context

#### Where should the reviewer start?

lib/index.js

#### How should this be manually tested?

All the functionality of Snyk CLI should remain intact.

We have a highly problematic module, "lib/index.js" aka "snyk".
It's stateful, it's a proxy for other modules, it introduces a lot of import circles.

#### Background context

The import cycles we have at the moment are:
```
1) lib/index.js > lib/capture.js
2) lib/index.js > lib/monitor.js > lib/analytics.js
3) lib/analytics.js > lib/request/index.js > lib/request/request.js
4) lib/index.js > lib/monitor.js
5) lib/index.js > lib/snyk-test/index.js > lib/snyk-test/npm/index.js
6) lib/index.js > lib/snyk-test/index.js > lib/snyk-test/run-test.js
7) lib/index.js > lib/snyk-test/index.js > lib/snyk-test/run-test.js > lib/plugins/index.js > lib/plugins/npm/index.js
8) lib/index.js > lib/snyk-test/index.js > lib/snyk-test/run-test.js > lib/plugins/index.js > lib/plugins/yarn/index.js
```

This PR is a first baby step towards untangling index.js. It replaces "snyk.monitor" references with direct imports of "monitor", thus expanding the cycle 4 to:

```lib/index.js > lib/capture.js > lib/monitor.js```

Removing other cyclic imports will follow.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "build": "tsc",
     "build-watch": "tsc -w",
     "eslint": "eslint -c .eslintrc src",
+    "find-circular": "npm run build && madge --circular ./dist",
```
Comment (miiila) [inline package.json]: Registry uses  `madge --extensions js,ts --circular ./src ./test`, not sure if there's a benefit of doing that, just sharing :-)

File: src/lib/index.js [hunk 2]
```diff
@@ -3,6 +3,14 @@ module.exports = snyk;
 var cluster = require('cluster');
 var snykConfig = require('./config');
 
+// This module exports a function with properties attached.
+// The properties represent other modules - so, it's some kind of "world object"
+// that is used to indirectly import modules.
+// The function modifies the object and returns it.
+// This also indtroduces some circular imports.
```
Comment (miiila) [inline src/lib/index.js]: typo, `introduces` (sorry for nitpicking)

File: src/lib/monitor.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module.exports = monitor;
+export = monitor;
```
Comment (miiila) [inline src/lib/monitor.ts]: Do we want to keep this or shall we `export function monitor` on line 40?

File: package.json (hunk 1 above)
Comment (kyegupov) [reply package.json]: Yeah, I've copied this from the registry. It looks like madge is not 100% comfy with Typepscript, so the result was that the cycles were found twice - once in src, once in dist.

File: src/lib/monitor.ts (hunk 3 above)
Comment (kyegupov) [reply src/lib/monitor.ts]: Hmm. While we're at it, we can patch all the clients, I guess...

**Pull request merged**